MATLAB/main.m

```diff
@@ -1,74 +1,49 @@
 clearvars; clc;
+config()  % Initialize const
+
 
-global const
-global truth_trajectory
-global computer_state_trajectory
-global actuators_trajectory
 
-addpath('utl');
-addpath('environmental_models');
-addpath('environmental_models/helper_functions');
+global const
+global main_state_trajectory
+global computer_state_follower_trajectory
+global computer_state_leader_trajectory
 
 dt=double(const.dt) * 1e-9;
-t_max = 2000;%0.01 * 90.0 * 60.0;  % Amount of time simulated (s)
-t_int = 10.0;               % Sampling interval        (s)
-%arrays to plot
+t_max = 2000;% Amount of time simulated (s)
+t_int = 10.0;% Sampling interval        (s)
 num_steps = floor(t_max/dt);
 sample_rate = t_int/dt;
 N = floor(num_steps /sample_rate);  % Number of samples
-config()  % Initialize the simulation
-
-[truth,actuators,sensor_state,computer_state]=initialize_states(1,'not_detumbled');
-computer_state_trajectory = cell(1,N);
-sensor_state_trajectory{1} = sensor_state;
-actuators_trajectory = cell(1,N);
-actuators_trajectory{1} = actuators;
-truth_trajectory = cell(1,N);
-truth_trajectory{1} = truth;
+main_state= initialize_main_state(1,'not_detumbled');
+[computer_state_follower,computer_state_leader]= initialize_computer_states('not_detumbled');
+main_state_trajectory = cell(1,N);
+main_state_trajectory{1} = main_state;
+computer_state_follower_trajectory = cell(1,N);
+computer_state_follower_trajectory{1} = computer_state_follower;
+computer_state_leader_trajectory = cell(1,N);
+computer_state_leader_trajectory{1} = computer_state_leader;
 t_s = 0.0;  % Timestamp of last data point
 n   = 1;     % Previously logged data index
 
 for step= 1:num_steps
-    %sense truth
-    tic;
-    sensor_readings = sensor_reading(sensor_state,truth,actuators);
-    t_sensors = toc;
+    sensor_readings_follower = sensor_reading(main_state.follower,main_state.leader);
+    sensor_readings_leader = sensor_reading(main_state.leader,main_state.follower);
     %update dynamics
-    tic;
-    truth = orbit_attitude_update_ode2(truth,actuators,dt);
-    t_orbit_att = toc;
-    %update time
-    truth.mission_time = truth.mission_time+const.dt;
-    truth.time = double(truth.mission_time)*1E-9;
-    %update sensor state (for example biases)
-    tic;
-    sensor_state_update(sensor_state,truth,dt);
-    t_sensor_state = toc;
-    %simulate flight computer
-    tic;
-    [computer_state,actuator_commands] = update_FC_state(computer_state,sensor_readings);
-    t_fc = toc;
+    main_state = main_state_update(main_state);
+    %simulate flight computers
+    [computer_state_follower,actuator_commands_follower] = update_FC_state(computer_state_follower,sensor_readings_follower);
```

Comment:
Could you clarify, again, what the update_FC_state function does? How should this play with sim?

Reply:
Correct me if I'm wrong @nhz2, but this should be an actual call to the teensy during a HITL. But for the MATLAB standalone version we'll have the cut down state machine Nathan has going (mainly just handles ADCS states).

Reply:
Since the Teensy can't handle all state values yet, how can we make this function handle partial state updates from the Teensy?

Reply:
Correct me if I'm wrong @nhz2, but this should be an actual call to the teensy during a HITL. But for the MATLAB standalone version we'll have the cut down state machine Nathan has going (mainly just handles ADCS states).

@tanishqaggarwal  This may be a conversation to have some other place, but what can the flight computer handle at his time?

Reply:
There's no controllers yet written for the flight controller, so there's no state reported yet by the flight controller.